Add a walkthrough to setup Lighthouse with Tekton

eddycharly opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

This PR adds a walkthrough to setup Tekton and Lighthouse from scratch on a kind cluster.

You can preview the walkthrough here https://github.com/eddycharly/dashboard/blob/walkthrough-lighthouse/docs/walkthrough/walkthrough-lighthouse.md

/cc @AlanGreene 
/cc @abayer 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

eddycharly commented:
/hold

( the pipeline run is not the good one )

a-roberts commented:
```

> Lighthouse is a lightweight ChatOps based webhook handler which can trigger Jenkins X Pipelines and soon Tekton Pipelines based on webhooks from multiple git providers such as: GitHub, GitHub Enterprise, BitBucket Server, BitBucket Cloud, GitLab, Gitea etc
```

sounds great!

abayer commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,421 @@
+
+From the [Lighthouse GitHub page](https://github.com/jenkins-x/lighthouse):
+
+> Lighthouse is a lightweight ChatOps based webhook handler which can trigger Jenkins X Pipelines and soon Tekton Pipelines based on webhooks from multiple git providers such as: GitHub, GitHub Enterprise, BitBucket Server, BitBucket Cloud, GitLab, Gitea etc
```
`s/and soon Tekton Pipelines/and Tekton Pipelines/`, and I'd remove the `BitBucket Cloud` and `Gitea` from the list - thanks for reminding me that Lighthouse's README needs to be updated... =)

eddycharly replied:
fixed, thanks

abayer commented:
Except for the things that I should have already updated in Lighthouse's README, this looks great! =)

eddycharly commented:
I'm sure i could have given more chatops examples but... at least it shows how to setup everything. How to use it is another story ;)

abayer commented:
@eddycharly This is unquestionably better documentation than I've ever written, so I can't complain. =)

eddycharly commented:
/hold cancel

I've set the correct pipeline run definition

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+> Lighthouse is a lightweight ChatOps based webhook handler which can trigger Jenkins X Pipelines and Tekton Pipelines based on webhooks from multiple git providers such as: GitHub, GitHub Enterprise, BitBucket Server, GitLab
+
+That is, Lighthouse brings ChatOps over a repository so that you can interact with a bot by writing comments over pull requests.
```
```suggestion
That is, Lighthouse brings ChatOps to a repository so that you can interact with a bot by writing comments on pull requests.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+That is, Lighthouse brings ChatOps over a repository so that you can interact with a bot by writing comments over pull requests.
+
+In addtion, it receives webhooks and triggers jobs to be executed according to the received changes, it then reports the job execution status and can merge pull requests automatically when all jobs succeed.
```
```suggestion
In addition, it receives webhooks and triggers jobs to be executed according to the received changes, it then reports the job execution status and can merge pull requests automatically when all jobs succeed.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+In addtion, it receives webhooks and triggers jobs to be executed according to the received changes, it then reports the job execution status and can merge pull requests automatically when all jobs succeed.
+
+Lighthouse is part of the jenkins-x platform but can be deployed in stand alone and plugged with Tekton pipelines easily.
```
```suggestion
Lighthouse is part of the jenkins-x platform but can be deployed stand-alone and plugged with Tekton pipelines easily.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+## Creating a GitHub OAuth token
+
+Before installing Lighthouse, you will need to create a OAuth token on your GitHub account.
```
```suggestion
Before installing Lighthouse, you will need to create an OAuth token on your GitHub account.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+To create the OAuth token on GitHub, browse https://github.com/settings/tokens and click the `Generate new token` button.
+
+Fill in the form and click the `Register token` button:
+- `Note`: put whatever let's you remember how this token is used (`lighthouse` might be a good hint)
```
```suggestion
- `Note`: put whatever helps you remember how this token is used (`lighthouse` might be a good hint)
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+![Creating a GitHub Oauth token](./walkthrough-lighthouse-oauth-token.png)
+
+**NOTE:** for those curious people, this token is not valid anymore ;-)
```
😆

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+In this step you will define a few environment variables that will be reused in the other steps below.
+
+Those environment variables will hold information related to the GitHub org/repo you want to connect with Lighthouse and the necessary information to let Lighthouse authenticate with GitHub.
```
```suggestion
These environment variables will hold information related to the GitHub org/repo you want to connect with Lighthouse and the necessary information to let Lighthouse authenticate with GitHub.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+| -------- | --- | ---------- |
+| USER | used by Lighthouse to interact with your repository | you know it |
+| REPOSITORY | the repository you want to connect with Lighthouse and that will send events | you know it |
+| OAUTH_TOKEN | the token used by Lighthouse to connect to your github repository | you generated one in [Creating a GitHub OAuth token](#creating-a-github-oauth-token) |
```
```suggestion
| OAUTH_TOKEN | the token used by Lighthouse to connect to your GitHub repository | you generated one in [Creating a GitHub OAuth token](#creating-a-github-oauth-token) |
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+| OAUTH_TOKEN | the token used by Lighthouse to connect to your github repository | you generated one in [Creating a GitHub OAuth token](#creating-a-github-oauth-token) |
+| NGROK_ENDPOINT | the ngrok endpoint connected to your local cluster ingress controller | you obtained it when [Exposing your cluster ingress controller over ngrok](#exposing-your-cluster-ingress-controller-over-ngrok) |
+| LIGHTHOUSE_NAMESPACE | the namespace where Lighthouse will be installed | you can choose whatever you want |
+| HMAC_TOKEN | a secret that will be shared between Lighthouse and GitHub webhooks | you can choose whatever you want, generate a random secret is fine |
```
```suggestion
| HMAC_TOKEN | a secret that will be shared between Lighthouse and GitHub webhooks | you can choose whatever you want, generating a random secret is fine |
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+| LIGHTHOUSE_NAMESPACE | the namespace where Lighthouse will be installed | you can choose whatever you want |
+| HMAC_TOKEN | a secret that will be shared between Lighthouse and GitHub webhooks | you can choose whatever you want, generate a random secret is fine |
+
+In my case, my GitHub user name being `eddycharly`, the repository to be used in this walkthrough being `tekton-ci`, and from the steps above, the script will be this one:
```
```suggestion
For example, with the GitHub user name being `eddycharly`, the repository to be used in this walkthrough being `tekton-ci`, from the steps above the script will be:
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
````diff
@@ -0,0 +1,461 @@
+HMAC_TOKEN=$(LC_CTYPE=C tr -dc 'A-F0-9' < /dev/urandom | head -c42)
+```
+
+Please run the script above with your specific informations before continuing.
````
```suggestion
Please run the script above with your specific details before continuing.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+Configuring Lighthouse can be tricky, one approach is store config files in a git repository and use `config-updater` plugin to update the configuration when something changes in the repository.
+
+While this works i'm not a big fan of this approach, in this walkthrough you will manage the config files yourself, there are plenty of simpler solutions for managing them, especially when using helm as a deployment tool.
```
```suggestion
In this walkthrough you will manage the config files yourself, there are plenty of simpler solutions for managing them, especially when using helm as a deployment tool.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
````diff
@@ -0,0 +1,461 @@
+```
+
+Those two files are enough to get Lighthouse running in the context of this walkthrough.
+However, you can tweak the files content to get the behaviour that suits your needs, please refer to the documentation.
````
```suggestion
However, you can tweak the content to get the behaviour that suits your needs, please refer to the documentation.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+Lighthouse jobs are configured using a centralized approach where jobs are defined in the `config.yaml` file.
+
+There are mainly two categories of jobs `presubmits` that run against pull requests and `postsubmits` that run against pushes.
```
```suggestion
There are two main categories of jobs: `presubmits` that run against pull requests, and `postsubmits` that run against pushes.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+There are mainly two categories of jobs `presubmits` that run against pull requests and `postsubmits` that run against pushes.
+
+The `config.yaml` contains `presubmits` and `postsubmits` jobs definitions for every repositories that send notifications.
```
```suggestion
The `config.yaml` contains `presubmits` and `postsubmits` job definitions for every repository that sends notifications.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
````diff
@@ -0,0 +1,461 @@
+EOF
+```
+
+**NOTES:** the Lighthouse tekton controller listens in the deployed namespace so `pod_namespace` and `prowjob_namespace` must be set to the namespace where Lighthouse was deployed.
````
```suggestion
**NOTES:** the Lighthouse Tekton controller listens in the deployed namespace so `pod_namespace` and `prowjob_namespace` must be set to the namespace where Lighthouse was deployed.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+## Lighthouse running jobs when opening a pull request
+
+Pull requests opened on your repository will now send events to Lighthouse, Lighthouse will trigger the `PipelineRun`s defined in the `presubmit` jobs for the given repository (you can browse the Tekton Dashboard at `http://tekton-dashboard.127.0.0.1.nip.io/`).
```
```suggestion
Pull requests opened on your repository will now send events to Lighthouse, Lighthouse will trigger the `PipelineRun`s defined in the `presubmit` jobs for the given repository. You can browse the Tekton Dashboard at `http://tekton-dashboard.127.0.0.1.nip.io/`.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+Posting `/approve` in the pull request comments will add the `approved` label to the pull request and Lighthouse will merge the PR.
+
+**Different plugins bring different functionnalities**
```
```suggestion
**Different plugins bring different functionality**
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+**Different plugins bring different functionnalities**
+
+The Lighthouse functionnalities are plugin driven. You can add/remove functionnalities by changing the `plugins.yaml` file, configuring plugins and enabling them for individual repositories.
```
```suggestion
The Lighthouse functionality is plugin driven. You can add/remove functionality by changing the `plugins.yaml` file, configuring plugins and enabling them for individual repositories.
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+The Lighthouse functionnalities are plugin driven. You can add/remove functionnalities by changing the `plugins.yaml` file, configuring plugins and enabling them for individual repositories.
+
+You can read more about the Lighthouse configuration options on [Lighthouse GitHub page](https://github.com/jenkins-x/lighthouse).
```
```suggestion
You can read more about the Lighthouse configuration options on the [Lighthouse GitHub page](https://github.com/jenkins-x/lighthouse).
```

AlanGreene commented on docs/walkthrough/walkthrough-lighthouse.md:
```diff
@@ -0,0 +1,461 @@
+
+Remove the webhooks from your GitHub repository.
+
+Turn of the ngrok tunnel.
```
```suggestion
Turn off the ngrok tunnel.
```

AlanGreene reviewed (changes requested):
Looks great, had fun trying out Lighthouse and looking forward to using it more.

eddycharly commented:
Thanks @AlanGreene , i will address your comments later.
For now i need to update the walkthrough with recent changes in Lighthouse that should simplify the walkthrough a bit.

/hold

eddycharly commented:
@AlanGreene i addressed all your feedback, thanks.

I also updated the walkthrough with the latest changes to Lighthouse (this walkthrough does not need `kubectl` anymore).

eddycharly commented:
/hold cancel
